framework/src/userobjects/UserObject.C

```diff
@@ -89,9 +89,6 @@ UserObject::UserObject(const InputParameters & parameters)
                "A user object may be specified as executing before or after "
                "AuxKernels, not both.");
 
-  mooseAssert(_sys.varKind() == Moose::VAR_SOLVER,
-              "We expect the system to conceptually be nonlinear or linear.");
```

Comment:
this is for a solution user object (or similar?) that would be working on an auxvariable

Reply:
I can raise this error to the less abstract levels for sure. But at this point PF was using user objects on auxkernels and we just didn't see it because we defaulted to the 0th solversystem at object creation.

Reply:
it s fine for me to allow UOs on auxvariables.

Reply:
@lindsayad , hey we needed this to be able to do BCs on auxvariables. What is your opinion on this? It is mostly the changes in the FEProblemBase class.

Reply:
I guess it's fine if nothing else is breaking